server/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesIndexResponseTests.java

```diff
@@ -81,34 +85,68 @@ public void testShareResponsesUsingMappingHash() throws Exception {
             inList.add(new FieldCapabilitiesIndexResponse(index, hashing, fieldCaps, canMatch));
         }
         Randomness.shuffle(inList);
-        final List<FieldCapabilitiesIndexResponse> serializedList;
-        try (BytesStreamOutput output = new BytesStreamOutput()) {
-            FieldCapabilitiesIndexResponse.writeList(output, inList);
-            try (
-                StreamInput in = new NamedWriteableAwareStreamInput(
-                    output.bytes().streamInput(),
-                    new NamedWriteableRegistry(Collections.emptyList())
-                )
-            ) {
-                serializedList = FieldCapabilitiesIndexResponse.readList(in);
+        // old versions
+        {
+            final Version oldVersion = VersionUtils.randomPreviousCompatibleVersion(random(), Version.V_8_2_0);
```

Comment:
I wonder if this will always return what we expect, because the lower bound is `minIndexCompatVersion` but we'd want `minCompatVersion` instead?

Reply:
Yes, we need a compact version. I fixed this in https://github.com/elastic/elasticsearch/pull/84596/commits/8657d34012fe4211276183d6a077940070cd8ac2.